src/main/java/com/denizenscript/depenizen/bukkit/objects/towny/TownTag.java

```diff
@@ -224,6 +226,30 @@ public static void registerTags() {
             return new ElementTag(object.town.getBoard());
         });
 
+        // <--[tag]
+        // @attribute <TownTag.get_rank[(<rank>)]>
+        // @returns ListTag
+        // @plugin Depenizen, Towny
+        // @description
+        // Returns a list of the town's members with a given rank. Players will be valid PlayerTag instances, non-players will be plaintext of the name.
+        // -->
+        tagProcessor.registerTag(ListTag.class, "get_rank", (attribute, object) -> {
+            String rank = attribute.getParam();
+            ListTag list = new ListTag();
+            for (Resident resident : object.town.getRank(rank)) {
+                if (resident.getUUID() != null) {
+                    OfflinePlayer pl = Bukkit.getOfflinePlayer(resident.getUUID());
+                    if (pl.hasPlayedBefore()) {
+                        list.addObject(new PlayerTag(pl));
+                        continue;
+                    }
+                }
+                list.add(resident.getName());
+            }
+            return list;
```

Comment:
Much of this code is repeated in the `outlaws` tag, Perhaps create a helper method for it?